lib/worldwide/region.rb

```diff
@@ -384,6 +385,18 @@ def province_optional?
 
     private
 
+    def iso_to_zone_code(iso_code)
+      return iso_code if iso_code.nil? || iso_code.length < 3
+
+      country_code, iso_code = iso_code.split("-")
+      @iso_code_zones ||= Worldwide::Regions.all.select(&:country?)
```

Comment:
I worry about the repeated cost of computing the set of all countries having subregions whose legacy code starts with `c.iso_code`. This is an instance method, so we're looking at computing this very often for a long-lived application.

Could we compute and cache this list of qualifying country iso_codes in the class?